fedimint-core/src/task.rs

```diff
@@ -31,23 +32,41 @@ type JoinError = anyhow::Error;
 #[error("deadline has elapsed")]
 pub struct Elapsed;
 
-#[derive(Debug, Default)]
+#[derive(Debug)]
 struct TaskGroupInner {
     /// Was the shutdown requested, either externally or due to any task
     /// failure?
     is_shutting_down: AtomicBool,
     #[allow(clippy::type_complexity)]
     on_shutdown: Mutex<Vec<Box<dyn FnOnce() -> BoxFuture<'static, ()> + Send + 'static>>>,
+    on_shutdown_tx: watch::Sender<bool>,
+    on_shutdown_rx: watch::Receiver<bool>,
```

Comment:
Why keep the `rx` side around? It can be re-created using [`Sender::subscribe`](https://docs.rs/tokio/1.29.1/tokio/sync/watch/struct.Sender.html#method.subscribe) at any time.